### DISCUSSION TITLE
does react-native builder bob support use_frameworks!() ?

### DISCUSSION THREAD

**Discussion by FahadAminShovon** [callstack/react-native-builder-bob#412] [Q&A]
## does react-native builder bob support use_frameworks!() ?

I am trying to add react native firebase in my example ios project.
react-native-firebase [recommends](Notes: React-Native-Firebase uses use_frameworks, which has compatibility issues with Flipper, Hermes & Fabric.

Flipper: use_frameworks [is not compatible with Flipper](https://rnfirebase.io/#altering-cocoapods-to-use-frameworks). You need to disable Flipper by commenting out the :flipper_configuration line in your Podfile.) using use_fireworks .
If I disable use_flipper the example project is unable to resolve the exported header files from module folder.

**Comment by yasintorun** [comment]
did you know any solution?

**Comment by descorp** [comment]
FWIW  two main issue I have faced:

1) that Xcode replaces hyphen with underscore:

```
#if __has_include(<MY-LIB-NAME/SomeClass.h>)
#import <MY-LIB-NAME/SomeClass.h>
#else
#import <MY_LIB_NAME/SomeClass.h> // this works for `use_frameworks!`
#endif
```

2) swift bridge header is under module now:

```
#if __has_include("MY_LIB_NAME-Swift.h")
#import "MY_LIB_NAME-Swift.h"
#else
#import "MY_LIB_NAME/MY_LIB_NAME-Swift.h" // this works for `use_frameworks!`
#endif
```
